src/biogeochem/CNBalanceCheckMod.F90

```diff
@@ -72,13 +86,63 @@ subroutine InitAllocate(this, bounds)
   end subroutine InitAllocate
 
   !-----------------------------------------------------------------------
-  subroutine BeginCNBalance(this, bounds, num_soilc, filter_soilc, &
+  subroutine BeginCNGridcellBalance(this, bounds, &
+       cnveg_carbonstate_inst, cnveg_nitrogenstate_inst, &
+       c_products_inst, n_products_inst)
+    !
+    ! !DESCRIPTION:
+    ! Calculate beginning gridcell-level carbon/nitrogen balance
+    ! for mass conservation check
+    !
+    ! Should be called after CN state summaries have been computed
+    ! and before the dynamic landunit area updates
+    !
+    ! !USES:
+    !
+    ! !ARGUMENTS:
+    class(cn_balance_type)         , intent(inout) :: this
+    type(bounds_type)              , intent(in)    :: bounds
+    type(cnveg_carbonstate_type)   , intent(in)    :: cnveg_carbonstate_inst
+    type(cnveg_nitrogenstate_type) , intent(in)    :: cnveg_nitrogenstate_inst
+    type(cn_products_type)         , intent(in)    :: c_products_inst
+    type(cn_products_type)         , intent(in)    :: n_products_inst
+    !
+    ! !LOCAL VARIABLES:
+    integer :: g
+    integer :: begg, endg
+    !-----------------------------------------------------------------------
+
+    associate(                                            &
+         grc_begcb    => this%begcb_grc                  , & ! Output: [real(r8) (:)]  (gC/m2) gridcell carbon mass, beginning of time step
+         grc_begnb    => this%begnb_grc                  , & ! Output: [real(r8) (:)]  (gN/m2) gridcell nitrogen mass, beginning of time step
+         totgrcc      => cnveg_carbonstate_inst%totc_grc , & ! Input:  [real(r8) (:)]  (gC/m2) total gridcell carbon, incl veg and cpool
+         totgrcn      => cnveg_nitrogenstate_inst%totn_grc, & ! Input:  [real(r8) (:)]  (gN/m2) total gridcell nitrogen, incl veg
+         cropprod1c_grc => c_products_inst%cropprod1_grc , & ! Input:  [real(r8) (:)]  (gC/m2) carbon in crop products
+         cropprod1n_grc => n_products_inst%cropprod1_grc , & ! Input:  [real(r8) (:)]  (gC/m2) nitrogen in crop products
+         tot_woodprodc_grc => c_products_inst%tot_woodprod_grc, & ! Input:  [real(r8) (:)]  (gC/m2) total carbon in wood products
+         tot_woodprodn_grc => n_products_inst%tot_woodprod_grc & ! Input:  [real(r8) (:)]  (gC/m2) total nitrogen in wood products
```

Comment:
Going with both suggestions. However, I left the `c` and `n` unchanged in `totc, totn, begcb, begnb` because these variables are more widespread. For the sake of easier searching and finding of these variables, I think leaving them unchanged is better for now.

Reply:
Good point, thanks.